### PR TITLE
Fix bugs in RBM notebooks

### PR DESCRIPTION
**Pull request by pradnyeshjoshi**
### Description

1. RBM algorithm in its current implementation requires ratings to be integers. Added code to map float ratings to integers in RBM quick start and deep dive notebooks.
2. Minor latex change in RBM deep dive notebook so that the equations render properly.
3. Disabled TF warnings in RBM deep dive notebook.
4. Previously the batch size for precision calculation, which is a part of the training loop, was hardcoded and equal to the number of users (which caused OOM issue during model fitting for 10M and 20M movielens datasets). Changing this batch size resolves the OOM error during model fitting.

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: examples/00_quick_start/rbm_movielens.ipynb [hunk 1]
```diff
@@ -60,7 +60,7 @@
       "System version: 3.7.11 (default, Jul 27 2021, 14:32:16) \n",
       "[GCC 7.5.0]\n",
       "Pandas version: 1.3.4\n",
-      "Tensorflow version: 2.6.1\n"
+      "Tensorflow version: 1.15.5\n"
```
Comment (miguelgfierro) [inline examples/00_quick_start/rbm_movielens.ipynb:63]: hey @pradjoshi can you merge the latest stating to your branch, reinstall using TF 2 and rereun again?

Comment (pradnyeshjoshi) [reply examples/00_quick_start/rbm_movielens.ipynb]: hi @miguelgfierro I have reinstalled recommenders using TF2 and rerun the notebooks.

File: recommenders/models/rbm/rbm.py [hunk 2]
```diff
@@ -554,7 +554,7 @@ def train_test_precision(self, xtst):
 
             self.sess.run(
                 self.iter.initializer,
-                feed_dict={self.vu: xtst, self.batch_size: xtst.shape[0]},
+                feed_dict={self.vu: xtst, self.batch_size: self.minibatch},
```
Comment (pradnyeshjoshi) [inline recommenders/models/rbm/rbm.py:557]: Changing batch size here resolves OOM error during model fitting for 10M/ 20M movielens datasets.

**Review by miguelgfierro — APPROVED** [review]
@pradnyeshjoshi to me this looks ok to go, after the minor comments on logs. 

@anargyri do you want to include this in the release?

**Comment by anargyri** [comment]
> @pradnyeshjoshi to me this looks ok to go, after the minor comments on logs.
> 
> @anargyri do you want to include this in the release?

I don't know, maybe if there is time. First we need to resolve the tests that fail in the release.

**Comment by anargyri** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/microsoft/recommenders/pull/1581/discussion/) on ReviewNB
> 
> _miguelgfierro commented on 2021-12-15T08:54:38Z_ ----------------------------------------------------------------
> 
> Same comment as before with the verbosity

Yes, this is strange. Other algos like deeprec don't have this amount of output. Maybe you could check what the difference is from those.

**Review by anargyri — APPROVED** [review]

**Comment by anargyri** [comment]
> > View / edit / reply to [this conversation](https://app.reviewnb.com/microsoft/recommenders/pull/1581/discussion/) on ReviewNB
> > _miguelgfierro commented on 2021-12-15T08:54:38Z_ ----------------------------------------------------------------
> > Same comment as before with the verbosity
> 
> Yes, this is strange. Other algos like deeprec don't have this amount of output. Maybe you could check what the difference is from those.

I think I fixed it according to https://github.com/tensorflow/tensorflow/issues/18146

**Comment by pradnyeshjoshi** [comment]
<p>Looks like Andreas resolved this already!</p>

---

 <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1581/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pradnyeshjoshi** [comment]
<p>I have created a <a href="https://msazure.visualstudio.com/One/_sprints/taskboard/AGCI%20AI%20-%20Retail/One/28-Ni/2%20Wk%20Sprints/Sprint%2011?workitem=12829711" target="_blank">task</a> for it. Should I create an issue on GitHub instead?</p>

---

 <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1581/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pradnyeshjoshi** [comment]
<p>Andreas resolved this.</p>

---

 <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1581/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pradnyeshjoshi** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1581/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Pull request merged**